maven-settings-builder/src/main/java/org/apache/maven/settings/validation/DefaultSettingsValidator.java

```diff
@@ -42,11 +43,10 @@
 @Singleton
 public class DefaultSettingsValidator implements SettingsValidator {
 
-    private static final String ID_REGEX = "[A-Za-z0-9_\\-.]+";
+    private static final String ID = "[A-Za-z0-9_\\-.]+";
```

Comment:
Here, the dot isn't escaped, a few classes it wasn't. As far as I remember the dot has no special meeting in ranges.